service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -44,22 +44,10 @@
      */
     public abstract List<String> uris();
 
-    // hides implementation details
     public static Builder builder() {
-        return ImmutableServiceConfiguration.builder();
+        return new Builder();
     }
 
-    // hides implementation details
-    public interface Builder {
-
-        Builder apiToken(BearerToken apiToken);
-
-        Builder security(SslConfiguration security);
-
-        Builder uris(Iterable<String> uris);
-
-        Builder from(ServiceConfiguration otherConfig);
-
-        ServiceConfiguration build();
+    public static final class Builder extends ImmutableServiceConfiguration.Builder {
```

Comment:
Is there an email list or something like that where we can discuss more in depth? Although this method of using the generated class is interesting, philosophically it doesn't seem correct -- the codebase won't compile at all without the generated classes, and it breaks the idea that the code that we write specifies the API, and the generator classes just fill them out. Any generated method that happens to be public will "leak" into the API, and if `Immutables` later changes and either generates more public methods or changes the public methods that are generated in the dependent class, that will introduce external API breaks without any author action. I think the readability will suffer as well -- this isn't the standard way that frameworks like `Immutables` or `AutoValue` are used.